src/ILCompiler.ReadyToRun/src/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -738,5 +716,312 @@ private static bool IsDynamicStatics(TypeDesc type)
             }
             return false;
         }
+
+        private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESOLVED_TOKEN* pConstrainedResolvedToken, CORINFO_METHOD_STRUCT_* callerHandle, CORINFO_CALLINFO_FLAGS flags, CORINFO_CALL_INFO* pResult)
+        {
+#if DEBUG
+            // In debug, write some bogus data to the struct to ensure we have filled everything
+            // properly.
+            MemoryHelper.FillMemory((byte*)pResult, 0xcc, Marshal.SizeOf<CORINFO_CALL_INFO>());
+#endif
+            MethodDesc method = HandleToObject(pResolvedToken.hMethod);
+
+            // Spec says that a callvirt lookup ignores static methods. Since static methods
+            // can't have the exact same signature as instance methods, a lookup that found
+            // a static method would have never found an instance method.
+            if (method.Signature.IsStatic && (flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_CALLVIRT) != 0)
+            {
+                throw new BadImageFormatException();
+            }
+
+            // This block enforces the rule that methods with [NativeCallable] attribute
+            // can only be called from unmanaged code. The call from managed code is replaced
+            // with a stub that throws an InvalidProgramException
+            if (method.IsNativeCallable && (flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) == 0)
+            {
+                ThrowHelper.ThrowInvalidProgramException(ExceptionStringID.InvalidProgramNativeCallable, method);
+            }
+
+            TypeDesc exactType = HandleToObject(pResolvedToken.hClass);
+
+            TypeDesc constrainedType = null;
+            if ((flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_CALLVIRT) != 0 && pConstrainedResolvedToken != null)
+            {
+                constrainedType = HandleToObject(pConstrainedResolvedToken->hClass);
+            }
+
+            bool resolvedConstraint = false;
+            bool forceUseRuntimeLookup = false;
+            bool targetIsFatFunctionPointer = false;
```

Comment:
There's no fat function pointers in CoreCLR.

Reply:
I removed this variable so we just always say false for CPAOT.